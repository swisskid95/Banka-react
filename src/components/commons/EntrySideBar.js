/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { toggleMenu } from '../../utils/sideBarOpen';

const EntrySideBar = ({ welcomeText, mainText, buttonText }) => {
  return (
    <aside className="sidebar">
      <div onClick={toggleMenu} className="sidebar__close">
        &times;
      </div>
      <h1 className="sidebar__title">{welcomeText}</h1>
      <p className="sidebar__text">{mainText}</p>
      <Link to={`/${buttonText}`} className="sidebar__btn">
        {buttonText}
      </Link>
    </aside>
  );
};

EntrySideBar.propTypes = {
  welcomeText: PropTypes.string.isRequired,
  mainText: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired
};

export default EntrySideBar;
