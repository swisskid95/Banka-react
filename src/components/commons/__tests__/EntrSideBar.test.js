/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import EntrySideBar from '../EntrySideBar';

describe('EntrySideBar', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <EntrySideBar
        welcomeText="HELLO, Friend"
        mainText="Not yet a member, click on the button below to Join in the benefits."
        buttonText="sign-up"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
