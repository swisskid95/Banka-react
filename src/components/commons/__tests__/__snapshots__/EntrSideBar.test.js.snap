// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntrySideBar should render correctly 1`] = `
<aside
  className="sidebar"
>
  <div
    className="sidebar__close"
  >
    Ã—
  </div>
  <h1
    className="sidebar__title"
  >
    HELLO, Friend
  </h1>
  <p
    className="sidebar__text"
  >
    Not yet a member, click on the button below to Join in the benefits.
  </p>
  <Link
    className="sidebar__btn"
    to="/sign-up"
  >
    sign-up
  </Link>
</aside>
`;
